app/controllers/tracks/concepts_controller.rb

```diff
@@ -0,0 +1,47 @@
+class Tracks::ConceptsController < ApplicationController
+  before_action :use_track
+
+  allow_unauthenticated! :index, :show
+
+  def authenticated_index
+    use_concepts
+
+    if current_user.joined_track?(@track)
+      render action: "index/joined"
+    else
+      render action: "index/unjoined"
+    end
+  end
+
+  def external_index
+    use_concepts
+  end
+
+  def authenticated_show
+    use_concept
+
+    if current_user.joined_track?(@track)
+      render action: "show/joined"
+    else
+      render action: "show/unjoined"
+    end
+  end
+
+  def external_show
+    use_concept
+  end
+
+  private
+  def use_track
+    @track = Track.find(params[:track_id])
+    @user_track = current_user&.user_track_for(@track)
```

Comment:
Any opinion in finding the right place for `#user_track_for`? I think if we continue doing methods like this, we'll end up with a very large `User` class. Maybe we could talk about this in a separate issue.

Reply:
https://github.com/exercism/v3-website/issues/155 🙂

Reply:
I won't block merging on this for now, but will change it based on what we decide there.